metadata-ingestion/src/datahub/ingestion/transformer/system_metadata_transformer.py

```diff
@@ -20,9 +20,10 @@ def auto_system_metadata(
 
     for workunit in stream:
         if set_system_metadata:
-            workunit.metadata.systemMetadata = SystemMetadataClass(
-                lastObserved=get_sys_time(), runId=ctx.run_id
-            )
+            if not workunit.metadata.systemMetadata:
+                workunit.metadata.systemMetadata = SystemMetadataClass(runId=ctx.run_id)
+            if not workunit.metadata.systemMetadata.lastObserved:
```

Comment:
```suggestion
            if not workunit.metadata.systemMetadata.lastObserved is None:
```

Reply:
Good catch, in case somebody sets `lastObserved` explicitly to `0`, we need to remove the leading `not` though in such case :)